lib/share_progress/button.rb

```diff
@@ -4,15 +4,29 @@
 module ShareProgress
   class Button
 
+    attr_accessor :id, :page_url, :page_title, :button_template, :share_button_html, :is_active
+
     class << self
-      def create
+
+      def create(page_url, button_template, raw_options={})
+        created = update(nil, page_url, button_template, raw_options)
+        created.nil? ? nil : new_from_fields(created)
```

Comment:
I think this line is instantiating a new `Button` object from the created values if `created` is nil? It's not wholly clear to me. Maybe a spot to be a bit more verbose to explain what's going on.